catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/XMLInputTransformer.java

```diff
@@ -0,0 +1,318 @@
+        this.saxEventHandlerConfiguration = saxEventHandlerConfiguration;
+    }
+
+    @Override
+    public String getVersion() {
+        return version;
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public String getTitle() {
+        return title;
+    }
+
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public String getOrganization() {
+        return organization;
+    }
+
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public void setOrganization(String organization) {
+        this.organization = organization;
+    }
+
+    /**
+     * Set the {@link Metacard} type that should be created by this transformer
+     *
+     * @param metacardType a {@link MetacardType}
+     */
+    public void setMetacardType(MetacardType metacardType) {
+        this.metacardType = metacardType;
+    }
+
+    public MetacardType getMetacardType() {
+        return this.metacardType;
+    }
+
+    /**
+     * Defines and returns a {@link DynamicMetacardType} based on component Sax Event Handler Factories
+     * and what attributes they populate
+     *
+     * @return a DynamicMetacardType that describes the type of metacard that is created in this transformer
+     */
+
+    public DynamicMetacardType defineMetacardType() {
+        List<SaxEventHandlerFactory> filteredSaxEventHandlerFactories = saxEventHandlerFactories
+                .stream().filter(p -> saxEventHandlerConfiguration.contains(p.getId()))
+                .collect(Collectors.toList());
+        Set<AttributeDescriptor> attributeDescriptors = new HashSet<>();
+        for (SaxEventHandlerFactory factory : filteredSaxEventHandlerFactories) {
+            attributeDescriptors.addAll(factory.getSupportedAttributeDescriptors());
+        }
+
+        /*
+         * These two attributes (metadata and ID) are added to the metacard by the input transformer itself
+         */
+        attributeDescriptors
+                .add(new AttributeDescriptorImpl(Metacard.METADATA, true /* indexed */, true /* stored */,
+                        false /* tokenized */, false /* multivalued */, BasicTypes.XML_TYPE));
+        attributeDescriptors
+                .add(new AttributeDescriptorImpl(Metacard.ID, true /* indexed */, true /* stored */,
+                        false /* tokenized */, false /* multivalued */, BasicTypes.STRING_TYPE));
+        return new DynamicMetacardType(attributeDescriptors, id);
+    }
+
+    public void init() {
+        this.metacardType = defineMetacardType();
+    }
+
+    public void destroy(int code) {
```

Comment:
It's required for the Managed Service Factory. It's in the `blueprint.xml` of `catalog-transformer-streaming-impl`.